Applications/spire/source/common/local_technicals_model.cpp

```diff
@@ -0,0 +1,105 @@
+#include "spire/spire/local_technicals_model.hpp"
+
+using namespace boost;
+using namespace boost::signals2;
+using namespace Nexus;
+using namespace Spire;
+
+LocalTechnicalsModel::LocalTechnicalsModel(Security security)
+    : m_security(security) {}
+
+void LocalTechnicalsModel::set_close(Money price) {
+  m_close = price;
+  m_close_signal(*m_close);
+  m_last_price = m_close;
+  m_last_price_signal(*m_last_price);
+}
+
+void LocalTechnicalsModel::update(const TimeAndSale& time_and_sale) {
+  if(!m_open.is_initialized()) {
```

Comment:
Emit signals after you have updated everything based on the ```time_and_sale```. This avoids inconsistencies in the model.

Reply:
So regardless of what changes just update everything?

Reply:
Only emit signals for values that have actually updated.

Reply:
Are there cases where the time and sales volume is 0?

Reply:
I can't think of any, how will this impact your solution?

Reply:
You were saying only emit signals if the values changed so I assumed you meant the last price, which may be the same as the previous last price.

I was just wondering if there was special cases where a time and sale could happen without a change in volume.

Reply:
It's safe to assume that all time and sales have a quantity greater than 0.